cloud/docker/docker_image.py

```diff
@@ -1,457 +1,502 @@
-  name:
+  force:
     description:
-       - Image name to work with
-    required: true
-    default: null
-    aliases: []
-  tag:
+      - Use with absent state to un-tag and remove all images matching the specified name. Use with states 'present'
+        and 'tagged' to take action even when an image already exists. If archive_path is specified, the force option
+        will cause an existing archive to be overwritten.
+    default: false
+  http_timeout:
     description:
-       - Image tag to work with
+      - Timeout for HTTP requests during the image build operation. Provide a positive integer value for the number of
+        seconds.
+    default: null
     required: false
-    default: "latest"
-    aliases: []
-  nocache:
+  name:
     description:
-      - Do not use cache with building
-    required: false
-    default: false
-    aliases: []
-  docker_url:
+      - "Image name. Name format will be one of: name, repository/name, registry_server:port/name.
+        When pushing or pulling an image the name can optionally include the tag by appending ':tag_name'."
+    required: true
+  path:
     description:
-      - URL of docker host to issue commands to
+      - Use with state 'present' to build an image. Will be the path to a directory containing the context and
+        Dockerfile for building an image.
+    aliases:
+      - build_path
+    default: null
     required: false
-    default: ${DOCKER_HOST} or unix://var/run/docker.sock
-    aliases: []
-  use_tls:
-    description:
-      - Whether to use tls to connect to the docker server.  "no" means not to
-        use tls (and ignore any other tls related parameters). "encrypt" means
-        to use tls to encrypt the connection to the server.  "verify" means to
-        also verify that the server's certificate is valid for the server
-        (this both verifies the certificate against the CA and that the
-        certificate was issued for that host. If this is unspecified, tls will
-        only be used if one of the other tls options require it.
-    choices: [ "no", "encrypt", "verify" ]
-    version_added: "2.0"
-  tls_client_cert:
+  push:
     description:
-      - Path to the PEM-encoded certificate used to authenticate docker client.
-        If specified tls_client_key must be valid
-    default: ${DOCKER_CERT_PATH}/cert.pem
-    version_added: "2.0"
-  tls_client_key:
+      - "Use with state present to always push an image to the registry. The image name must contain a repository
+        path and optionally a registry. For example: registry.ansible.com/user_a/repository"
+    default: false
+  pull:
     description:
-      - Path to the PEM-encoded key used to authenticate docker client. If
-        specified tls_client_cert must be valid
-    default: ${DOCKER_CERT_PATH}/key.pem
-    version_added: "2.0"
-  tls_ca_cert:
+      - When building an image downloads any updates to the FROM image in Dockerfiles.
+    default: true
+  rm:
```

Comment:
Here is all that docker-py offers:  Downloads any updates to the FROM image in Dockerfiles.  I don't know how it interacts with cache. Seem like updating would invalidate the cache.